mitodl/open-edx-plugins
# feat: add edx-sysadmin plugin app

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/edx-sysadmin/issues/76

**Note 1:**

I have also ported over the tests along with other components. The tests are currently not running because we will need to configure the test runup for the overall open-edx-plugins repo at some point. And these tests would run then.

**Note 2:**
For the initial pass, I ignored most of the warnings with `# noqa:` (No QA) to keep the scope of the PR narrow and try to not break any working functionality. Once we are done with this and everything works smoothly we should try and fix some of the pre-commit errors especially `PTH***` ones.

#### What's this PR do?
- Ports over the https://github.com/mitodl/edx-sysadmin to open-edx-plugins
- Adds the required pants configurations to support package creation of sysadmin
- Adds required plugin settings so that the sysadmin behaves as a plugin within the edX context
- Fixes lots of pre-commit warnings/errors. (There were so many pre-commit issues since this is an old code and was introduced to pre-commit for the first time. I have ignored most of the warnings and fixed the straightforward ones.)

#### How should this be manually tested?
- First of all, install sysadmin as mentioned in https://github.com/mitodl/edx-sysadmin
- Test the features mentioned in https://github.com/mitodl/edx-sysadmin
- Now, Setup up `open-edx-plugins` and generate package build e.g. using `./pants package ::`
- Install the package generated from this repo over the existing installation and see that:
    - There are no duplicate sysadmin packages
    - Everything(User creation, Course deletion, Course addition, Git logs, sysadmin commands such as `git_add_course`) works the same, especially the Git logs and course details.

**Note: (PRs you can take a look at for the steps to test the sysadmin overall, Follow instructions on those PRs)**
- https://github.com/mitodl/edx-sysadmin/pull/31
- https://github.com/mitodl/edx-sysadmin/pull/29
- https://github.com/mitodl/edx-sysadmin/pull/23

#### Where should the reviewer start?
- From an existing installation from the [old repo](https://github.com/mitodl/edx-sysadmin).

#### Screenshots


<img width="1035" alt="image" src="https://github.com/mitodl/open-edx-plugins/assets/34372316/412f2327-13ae-43a7-8403-70d9ddc7ce7b">

<img width="1357" alt="image" src="https://github.com/mitodl/open-edx-plugins/assets/34372316/342e59be-41fb-4118-923b-9e3678a4dc1a">

<img width="1277" alt="image" src="https://github.com/mitodl/open-edx-plugins/assets/34372316/4f837f65-ee62-4eee-95fe-b0afdcae0aaf">

<img width="1139" alt="image" src="https://github.com/mitodl/open-edx-plugins/assets/34372316/dbf7d002-fee6-4b5f-98d5-9d6dce90736a">
